tutorials/RandomNumberGeneration/Tests.qs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+//////////////////////////////////////////////////////////////////////
+// This file contains testing harness for all tasks.
+// You should not modify anything in this file.
+//////////////////////////////////////////////////////////////////////
+
+namespace Quantum.Kata.RandomNumberGeneration {
+    
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Math;
+    open Quantum.Kata.Utils;
+    open Microsoft.Quantum.Arrays;
+    
+
+    // Exercise 1.
+    operation T1_RandomBit_Test () : Unit {
+        Message("Testing...");
+        mutable count = 0;
+
+        ResetOracleCallsCount();
+        for (N in 1..1000) {
+            set count += RandomBit();
+        }
+        if (count < 465) {
+            fail $"Unexpectedly low number of 1's generated. Generated {count} out of 1000";
+        } elif (count > 535) {
+            fail $"Unexpectedly high number of 1's generated. Generated {count} out of 1000";
+        }
+        CheckRandomCalls();
+    }
+
+    // Exercise 2.
+    operation T2_RandomTwoBits_Test () : Unit {
+        Message("Testing...");
+        CheckFlatDistribution(RandomTwoBits_Wrapper, 2, 1, 2, 200);
+    }
+
+    operation RandomTwoBits_Wrapper (throwaway: Int) : Int {
+        return RandomTwoBits();
+    }
+    
+    // Exercise 3.
+    operation T3_RandomNBits_Test () : Unit {
+        Message("Testing N = 1...");
+        CheckFlatDistribution(RandomNBits, 1, 0, 2, 465);
+        Message("Testing N = 2...");
+        CheckFlatDistribution(RandomNBits, 2, 1, 2, 200);
+        Message("Testing N = 10...");
+        CheckFlatDistribution(RandomNBits, 10, 461, 563, 0);
+
+    }
+
+    // f: operation to be tested
+    // numBits: number of bits in each result
+    // lowRange, highRange: median and average for generated dataset should be within [lowRange, highRange]
+    // minimum: the minimum number of times each possible number should be generated
+    operation CheckFlatDistribution (f : (Int => Int), numBits: Int, lowRange: Int, highRange: Int, minimum: Int) : Unit {
+        let max = PowI(2, numBits);
+        mutable counts = ConstantArray(max, 0);
+        mutable average = 0;
+        let total = 1000;
+
+        ResetOracleCallsCount();
+        for (i in 1..total) {
+            let val = f(numBits);
+            set average = average + val;
+            set counts w/= val <- counts[val] + 1;
+        }
+        CheckRandomCalls();
+
+        set average = average / total;
```

Comment:
I'm not sure whether you mean integer division or floating-point division here? Integer division will round down here, it might be exactly what you intend but wanted to double-check.

Reply:
It was intended to be integer division, because that's all that was needed with larger N values. I refactored it to be floating-point though, which is better especially for the N=1 tests.